world/shared_task_manager.cpp

```diff
@@ -0,0 +1,1820 @@
+{
+	for (const auto &m : s->GetMembers()) {
+		SendSharedTaskMemberChange(m.character_id, s->GetDbSharedTask().id, player_name, true);
+	}
+}
+
+void SharedTaskManager::MakeLeaderByPlayerName(SharedTask *s, const std::string &character_name)
+{
+	auto new_leader = s->FindMemberFromCharacterName(character_name);
+	if (new_leader.character_id == 0) {
+		SendLeaderMessageID(s, Chat::Red, SharedTaskMessage::IS_NOT_MEMBER, {character_name});
+		return;
+	}
+
+	bool found_new_leader = false;
+
+	std::vector<SharedTaskMember> members = s->GetMembers();
+	for (auto                     &m: members) {
+		LogTasksDetail(
+			"[MakeLeaderByPlayerName] character_id [{}] m.character_id [{}]",
+			new_leader.character_id,
+			m.character_id
+		);
+
+		m.is_leader = false;
+
+		// destination character is in shared task, make swap
+		if (new_leader.character_id == m.character_id) {
+			found_new_leader = true;
+			LogTasksDetail(
+				"[MakeLeaderByPlayerName] shared_task_id [{}] character_name [{}]",
+				s->GetDbSharedTask().id,
+				character_name
+			);
+
+			m.is_leader = true;
+		}
+	}
+
+	if (found_new_leader) {
+		s->SetMembers(members);
+		SaveMembers(s, members);
+		SendSharedTaskMemberListToAllMembers(s);
+		SendMembersMessageID(
+			s, Chat::Yellow, SharedTaskMessage::PLAYER_NOW_LEADER,
+			{new_leader.character_name, s->GetTaskData().title}
+		);
+
+		for (const auto &dz_id : s->dynamic_zone_ids) {
+			auto dz = DynamicZone::FindDynamicZoneByID(dz_id);
+			if (dz) {
+				dz->SetNewLeader(static_cast<uint32_t>(new_leader.character_id));
+			}
+		}
+	}
+}
+
+void SharedTaskManager::SaveMembers(SharedTask *s, std::vector<SharedTaskMember> members)
+{
+	std::vector<SharedTaskMembersRepository::SharedTaskMembers> dm = {};
+	dm.reserve(members.size());
+	for (auto &m: members) {
+		auto e = SharedTaskMembersRepository::NewEntity();
+
+		e.character_id   = m.character_id;
+		e.is_leader      = (m.is_leader ? 1 : 0);
+		e.shared_task_id = s->GetDbSharedTask().id;
+
+		dm.emplace_back(e);
+	}
+
+	SharedTaskMembersRepository::DeleteWhere(*m_database, fmt::format("shared_task_id = {}", s->GetDbSharedTask().id));
+	SharedTaskMembersRepository::InsertMany(*m_database, dm);
+}
+
+void SharedTaskManager::InvitePlayerByPlayerName(SharedTask *s, const std::string &player_name)
+{
+	auto character = CharacterDataRepository::GetWhere(
+		*m_database,
+		fmt::format("`name` = '{}' LIMIT 1", EscapeString(player_name))
+	);
+
+	auto character_id = !character.empty() ? character.front().id : 0;
+
+	// we call validation even for an invalid player so error messages occur
+	if (CanAddPlayer(s, character_id, player_name, false)) {
+		// send dialogue window
+		SendSharedTaskInvitePacket(s, character_id);
+
+		// keep track of active invitations at world
+		QueueActiveInvitation(s->GetDbSharedTask().id, character_id);
+	}
+}
+
+void SharedTaskManager::SendSharedTaskInvitePacket(SharedTask *s, int64 invited_character_id)
```

Comment:
`invited_character_id` should probably be uint32 to match other methods and cut down on narrowing conversion warnings

Reply:
Yeah warnings are one thing but kinda wanted to keep these wider for whenever we may change character_id to int64. Could go either way here